### PR TITLE
ksys::eco::LevelSensor::calculatePoints

### PR DESCRIPTION
**Pull request by alex-botw**
implemented calculatePoints going to continue scaleWeapon and scaleActor

---
This change is [Reviewable](https://reviewable.io/reviews/zeldaret/botw/90)

### CODE REVIEW

File: src/KingSystem/Ecosystem/ecoLevelSensor.cpp [hunk 1]
```diff
@@ -20,4 +21,55 @@ void LevelSensor::init(sead::Heap* heap) {
     mRootIter = new (heap) al::ByamlIter(res->getRawData());
 }
 
+void LevelSensor::calculatePoints()
+{
+    if (mDefaultPoints >= 0) {
+        mPoints = mDefaultPoints;
+    }
```
Comment (leoetlino) [inline src/KingSystem/Ecosystem/ecoLevelSensor.cpp:28]: considering returning early here to reduce unnecessary indentation

File: src/KingSystem/Ecosystem/ecoLevelSensor.cpp [hunk 2]
```diff
@@ -20,4 +21,55 @@ void LevelSensor::init(sead::Heap* heap) {
...
+    else {
+        al::ByamlIter iter;
+        if (!mRootIter->tryGetIterByKey(&iter, "flag")) {
+            return;
+        }
+        float mPointsSum = 0;
```
Comment (leoetlino) [inline src/KingSystem/Ecosystem/ecoLevelSensor.cpp:34]:
```suggestion
        float point_sum = 0;
```

File: src/KingSystem/Ecosystem/ecoLevelSensor.cpp [hunk 3]
```diff
@@ -20,4 +21,55 @@ void LevelSensor::init(sead::Heap* heap) {
...
+        for (int index = 0; index < iter.getSize(); index++) {
+            al::ByamlIter iterKill;
```
Comment (leoetlino) [inline src/KingSystem/Ecosystem/ecoLevelSensor.cpp:36]:
```suggestion
            al::ByamlIter iter_enemy;
```

File: src/KingSystem/Ecosystem/ecoLevelSensor.cpp (hunk 1 above)
Comment (alex-botw) [reply src/KingSystem/Ecosystem/ecoLevelSensor.cpp:29]: mismatch, code at the end

File: src/KingSystem/Ecosystem/ecoLevelSensor.cpp [hunk 4]
```diff
@@ -20,4 +21,55 @@ void LevelSensor::init(sead::Heap* heap) {
...
+            }
+            const char *name;
+            if (!iterKill.tryGetStringByKey(&name, "name")) {
+                return;
+            }
+            f32 kill_count;
+            s32 kill_mul = 0;
+            if (!gdt::Manager::instance()->getParam().get().getS32(&kill_mul, name)) {
+                bool non_skip = false;
+                bool work = gdt::Manager::instance()->getParam().get().getBool(&non_skip, name);
```
[the remark this answers is not in the document]
Comment (alex-botw) [reply src/KingSystem/Ecosystem/ecoLevelSensor.cpp:49]: bool non_skip = false;
                if (non_skip && gdt::Manager::instance()->getParam().get().getBool(&non_skip, name)) {
                    kill_mul = 1;
                }
same logic but mismatch

[the remark this answers is not in the document]
Comment (alex-botw) [reply src/KingSystem/Ecosystem/ecoLevelSensor.cpp:49]: even if swap non_skip and work

File: src/KingSystem/Ecosystem/ecoLevelSensor.cpp [hunk 5]
```diff
@@ -20,4 +21,55 @@ void LevelSensor::init(sead::Heap* heap) {
...
+            if (!iter.tryGetIterByIndex(&iterKill, index)) {
+                return;
+            }
+            const char *name;
+            if (!iterKill.tryGetStringByKey(&name, "name")) {
+                return;
+            }
+            f32 kill_count;
+            s32 kill_mul = 0;
+            if (!gdt::Manager::instance()->getParam().get().getS32(&kill_mul, name)) {
```
[the remark this answers is not in the document]
Comment (leoetlino) [reply src/KingSystem/Ecosystem/ecoLevelSensor.cpp:44]: I think you may have missed this comment

**Comment by alex-botw** [comment]
changed names and unique_kill part

**Review by leoetlino — APPROVED** [review]

**Pull request merged**